packages/api/cms-api/src/warnings/decorators/emit-warnings.decorator.ts

```diff
@@ -0,0 +1,33 @@
+import { type AnyEntity } from "@mikro-orm/postgresql";
+import { type CustomDecorator, SetMetadata, type Type } from "@nestjs/common";
+
+import { type WarningInput } from "../dto/warning.input";
+
+type EmitWarningsFunction<Entity extends AnyEntity = AnyEntity> = (item: Entity) => Promise<WarningInput[]>;
```

Comment:
I don't understand why "Input". Input to what?

Reply:
I thought about it as the input to create the warning but I changed it here to CreateWarningInput: 96d50a2231f2584e224555bc9dd79001217816a4

Reply:
It's odd that the result of a createWarnings is called "input", which would indicate that it's the input of the function. "Warning" would be more logical, but you can't use that, since it's not the warning entity. Maybe we could use `WarningData`, like MikroORM uses for create, assign, upsert, etc.?

https://mikro-orm.io/api/core/class/EntityManager#create

Reply:
i renamed it to WarningData: a797d782d8095dba5ed680c49b1f8af9aeb3b05f